packages/web_benchmarks/lib/src/benchmark_result.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A single metric value collected from the benchmark.
+class BenchmarkMetric {
+  /// Creates a benchmark metric.
+  BenchmarkMetric({
+    this.name,
+    this.value,
+  });
+
+  /// The name of the metric.
+  final String name;
+
+  /// The result of measuring a particular metric in this benchmark run.
+  final num value;
+
+  /// Serializes the benchmark metric to a JSON object.
+  Map<String, dynamic> toJson() {
+    return <String, dynamic>{
+      'name': name,
+      'value': value,
+    };
+  }
+}
+
+/// The result of running a benchmark.
+class BenchmarkResult {
+  /// Constructs a benchmark result.
+  BenchmarkResult(this.metrics);
+
+  /// All metrics collected in a benchmark run.
+  final List<BenchmarkMetric> metrics;
+
+  /// Serializes benchmark metrics to JSON.
+  List<Map<String, dynamic>> toJson() {
+    return metrics
+        .map<Map<String, dynamic>>((BenchmarkMetric e) => e.toJson())
+        .toList();
```

Comment:
The output format has changed. The new format is

```
[
  {
    "name": "scroll.html.preroll_frame.average",
    "value": 78.34693877551021
  },
  {
    "name": "scroll.html.preroll_frame.outlierAverage",
    "value": 847.5
  },
  {
    "name": "scroll.html.preroll_frame.outlierRatio",
    "value": 10.817270122427715
  },
  ...
]
```

The old format looks like

```
{
  "success": true,
  "data": {
    "scroll.html.preroll_frame.average": 93.88659793814433,
    "scroll.html.preroll_frame.outlierAverage": 1061.3333333333333,
    "scroll.html.preroll_frame.outlierRatio": 11.304417847077339,
    ...
  },
  "benchmarkScoreKeys": [
    "scroll.html.drawFrameDuration.average",
    "scroll.html.drawFrameDuration.outlierRatio",
    "scroll.html.totalUiFrame.average"
  ]
}

```

* Why is the new format preferred?
* Will it break any existing parts of the toolchain?
* How can we return a benchmark result representing an unsuccessful test? (Similar to the previous `TaskResult.failure`)
* How can we specify which parts are benchmarkScoreKeys?

Reply:
This is intentional. The old format was specific to devicelab. The devicelab expects information, such as whether the run was a success or not, encoded in the JSON message we send to it. The data is separated from `benchmarkScoreKeys` because the devicelab runs tests other than benchmarks, and so the results frequently contain data other than benchmark scores.

Since this is purely a benchmark package all we need to encode is benchmark scores, so we don't need to separate data from score keys. A failure is also easily expressible by simply throwing an exception and exiting the process with a non-zero exit code. So I simplified the output format by removing unnecessary features.

When we run this code in the devicelab, we will adapt it to the devicelab protocol (which is an internal implementation detail, and can change any time; it's not a public API). It's better to keep this package protocol-agnostic.

Reply:
You reminded me about the term "score". I've been struggling what to call the class I ended up calling `BenchmarkMetric`. I think a better name is `BenchmarkScore`. I'll rename.